src/main/java/seedu/address/storage/attendance/JsonAdaptedAttendanceRecordList.java

```diff
@@ -0,0 +1,84 @@
+package seedu.address.storage.attendance;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import javafx.util.Pair;
+import seedu.address.commons.core.index.Index;
+import seedu.address.commons.exceptions.IllegalValueException;
+import seedu.address.model.attendance.AttendanceRecord;
+import seedu.address.model.attendance.AttendanceRecordList;
+import seedu.address.model.attendance.Week;
+
+/**
+ * Jackson-friendly version of {@link AttendanceRecordList}.
+ */
+public class JsonAdaptedAttendanceRecordList {
```

Comment:
Just throwing out this alternative: perhaps placing the `Week` object here would be better instead of in the `JsonAdaptedAttendanceRecord` class. So instead of having a `List<JsonAdaptedAttendanceRecordList>` attribute, there can be a `Map<Week, JsonAdaptedAttendanceRecord>`. Jackson supports Json serialisation of `Map`s: https://www.baeldung.com/jackson-map. 

Purpose is for symmetry between the Model classes. And I think it may help in reducing lines of code and tests.

I think this current implementation is also suitable for purposes of this project.

Reply:
I did consider the direct de/serialisation of key-value pairs using the Jackson, however I was rather hesitant to use it as I was unsure how it would play with the current implementation of `JsonUtil`. Similarly, the `StudentAttendance` implementation was used to avoid the need to complicate the deserialisation of key-value pairs.

I do agree with you that the direct serialisation would be more intuitive and provide a one to one mapping with the actual classes in the model.